src/CMakeLists.txt

```diff
@@ -80,7 +80,11 @@ function(PINOCCHIO_TARGET target_name)
         PARENT_SCOPE)
   endif()
 
-  add_library(${LIB_NAME} ${LIBRARY_TYPE})
+  if(NOT "${LIBRARY_TYPE}" STREQUAL "INTERFACE")
+    add_library(${LIB_NAME} ${LIBRARY_TYPE} ${${PROJECT_NAME}_CORE_PUBLIC_HEADERS})
+  else()
+    add_library(${LIB_NAME} ${LIBRARY_TYPE})
+  endif()
   target_link_libraries(${LIB_NAME} ${LIBRARY_PUBLIC_SCOPE} ${PROJECT_NAME}_headers)
```

Comment:
You can maybe move the target_sources command call bellow:

```cpp
  if(NOT ARGS_INTERFACE)
    target_sources(${LIB_NAME} PRIVATE ${ARGS_SOURCES})
    set_target_properties(
      ${LIB_NAME}
      PROPERTIES LINKER_LANGUAGE CXX
                 INSTALL_RPATH "\$ORIGIN"
                 VERSION ${PROJECT_VERSION})
  endif()
  ```

Reply:
Super. Thanks for the suggestion.

Reply:
The idea was more to use the target_source command instead of moving the add_library command.
Also, the comment is still relevant. After CMake 3.16, it's possible to attach headers to an INTERFACE library.